### PR TITLE
fix: changed .modal-backdrop to .offcanvas-backdrop

### LINKED ISSUE #5967: Offcanvas is hidden behind modal-backdrop due to z-index, change to use offcanvas-backdrop
## Describe the bug

It looks like there is a new class for offcanvas specific backdrops, which should now be .offcanvas-backdrop instead of .modal-backdrop (https://github.com/twbs/bootstrap/pull/34149).

.modal-backdrop has a higher z-index (1050) than .offcanvas (1045) by default, and so the offcanvas sidebar is hidden behind the backdrop.

## Reproducible Example

https://codesandbox.io/s/friendly-ride-phnkw

## Environment (please complete the following information)

- Operating System: [e.g. macOS]: Windows
- Browser, Version [e.g. Chrome 74]: Chrome
- React-Bootstrap Version [e.g. 1.0.0-beta.8]: v2 beta 4
- Bootstrap Version: 5.1.0
## Additional context

Looks like changing modalPrefix 2nd arg to 'offcanvas' should do the trick @ https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Offcanvas.tsx#L224, happy to create a pr for this.

### PR DESCRIPTION
**Pull request by nsi319**
Fix #5967

### CODE REVIEW

File: src/Offcanvas.tsx [hunk 1]
```diff
@@ -221,7 +221,7 @@ const Offcanvas: BsPrefixRefForwardingComponent<'div', OffcanvasProps> =
       const handleHide = useEventCallback(onHide);
 
       bsPrefix = useBootstrapPrefix(bsPrefix, 'offcanvas');
-      const modalBsPrefix = useBootstrapPrefix(undefined, 'modal');
+      const modalBsPrefix = useBootstrapPrefix(undefined, 'offcanvas');
```
Comment (kyletsang) [inline src/Offcanvas.tsx:224]: We don't need `modalBsPrefix` anymore because it's essentially the same as the one above it.

Comment (nsi319) [reply src/Offcanvas.tsx:224]: Okay. Let me make the necessary changes.

**Comment by nsi319** [comment]
Hi @kyletsang. May I know why Install Dependencies at [checks](https://github.com/react-bootstrap/react-bootstrap/runs/3387615861) failed previously?

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks!

**Comment by kyletsang** [comment]
> 
> 
> Hi @kyletsang. May I know why Install Dependencies at [checks](https://github.com/react-bootstrap/react-bootstrap/runs/3387615861) failed previously?

It's just an intermittent failure.  Nothing to worry about

**Pull request merged**